### PR TITLE
Updating thriftrw to 3.x

### PR DESCRIPTION
**Pull request by malandrew**
This is ready for a first pass review. 

The only additional thing I want to change is making sure that you need to declare that the file system will be accessed. Please confirm that this is desirable.

### CODE REVIEW

File: tcollector/reporter.js [hunk 1]
```diff
@@ -25,8 +25,8 @@ var fs = require('fs');
 var assert = require('assert');
 var Buffer = require('buffer').Buffer;
 
-var tcollectorSpec =
-    fs.readFileSync(path.join(__dirname, 'tcollector.thrift'), 'utf8');
+var tcollectorSpecPath = path.join(__dirname, 'tcollector.thrift');
+var tcollectorSpec = fs.readFileSync(tcollectorSpecPath, 'utf8');
```
Comment (kriskowal) [inline tcollector/reporter.js]: tcollectorSpec is no longer used.

**Comment by malandrew** [comment]
I'm assuming that the default for TChannelAsThrift is to allowFileSystemAccess whereas the default for thriftrw.Thrift should be to disallow filesystem access.

**Comment by kriskowal** [comment]
@malandrew Interesting observation. We could infer allowFilesystemAccess = false if options.idls is provided to the root constructor.

Looks ok. Green the tests and :ok: to land.

**Comment by Raynos** [comment]
-1 breaks back compat.

Please write tests for both source and entryPoint

**Pull request closed without merging**